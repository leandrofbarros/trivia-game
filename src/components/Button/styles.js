import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    marginBottom: 30
  },
  title: {
    fontSize: 28,
    fontWeight: '400',
  },
})
