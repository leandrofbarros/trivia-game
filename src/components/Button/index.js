// @flow

import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import styles from './styles'
import type { NavigationState } from "react-navigation"

type Props = {
  title: string,
  onPress: () => void
}

const Button = ({ title, onPress }: Props) => (
  <View style={styles.container}>
    <TouchableOpacity onPress={onPress}>
      <Text style={styles.title}>{title}</Text>
    </TouchableOpacity>
  </View>
)

export default Button
