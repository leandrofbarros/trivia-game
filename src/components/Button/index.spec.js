// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import Button from './index'

describe('#Button', () => {
  it('Renders correctly', () => {
    const component = (
      <Button
        title="Test Button"
        onValueChange={() => null}
      />
    )

    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})