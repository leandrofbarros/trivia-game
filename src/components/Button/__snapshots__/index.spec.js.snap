// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Button Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 30,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 28,
          "fontWeight": "400",
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
